### PR TITLE
Update hacking documentation.

### PR DESCRIPTION
**Pull request by thomir**

### CODE REVIEW

File: doc/hacking.rst [hunk 1]
```diff
@@ -39,6 +39,18 @@ Assignment above) needs to be licensed under the `MIT license`_ that testtools
 uses, so that testtools can ship it.
 
 
+Installing
```
Comment (rbtcollins) [inline doc/hacking.rst]: I don't think installing makes sense *here* in the docs: this is docs on hacking, and the 'testing' section below covers this.

Secondly, for users, I think we want to say 'pip install testtools' - anyone installing from source should be coming from a position of knowledge after all...

**Comment by thomir** [comment]
@rbtcollins Good point.

I've updated the heading to 'Building', de-duplicated the pip installation instruction. This means that the 'Testing' section is now solely about running and writing tests, and the 'Building' section now contains all the pip etc. content.

Thoughts?

**Comment by rbtcollins** [comment]
Squashed and rebased and landed.

**Comment by thomir** [comment]
Thanks @rbtcollins  - I would have squashed the commits myself had you approved.

**Comment by rbtcollins** [comment]
That would have been fine too; I just didn't see any point in a hand-off once it was good :)

**Pull request merged**